lua/neotest-dotnet/vstest_wrapper.lua

```diff
@@ -259,18 +310,11 @@ function M.discover_tests(path)
     end, { type = "file", path = root, limit = math.huge })
 
     for _, project in ipairs(projects) do
-      local dir_name = vim.fs.dirname(project)
-      local proj_name = vim.fn.fnamemodify(project, ":t:r")
-
-      local proj_dll_path =
-        -- TODO: this might break if the project has been compiled as both Development and Release.
-        vim.fs.find(function(name)
-          return string.lower(name) == string.lower(proj_name .. ".dll")
-        end, { type = "file", path = dir_name })[1]
-
-      if proj_dll_path then
-        dlls[#dlls + 1] = proj_dll_path
-        local project_open_err, project_stats = nio.uv.fs_stat(proj_dll_path)
+      local local_proj_info = M.get_proj_info(project)
```

Comment:
The addition of this loop for large solutions with lots of projects leads to some pretty immense RAM usage (it is maxing out my 32gb) so we need to take this out and populate the cache some other way. Running multiple MSBuild processes in parallel is the problem I think.

Reply:
The test discovery appears to work fine without this eager population of the cache so I have just removed it.
Memory consumption appears stable on my test solutions, but they do not contain _that_ many projects